### PR TITLE
Try to improve OSCALLoaderForm structure

### PR DESCRIPTION
**Pull request by kylelaker**
Currently, the `onUrlChange()` is fired whenever any character is typed.
This is sorta undesirable because it can result in various re-rendering
happening in a few places prematurely.

As-is, this doesn't work. It seems like the actual value received by
`OSCALLoader` "lags" behind the input value by one form submission.

Relies on #490

### CODE REVIEW

**Comment by Bronstrom** [comment]
>Relies on https://github.com/EasyDynamics/oscal-react-library/pull/490

With 490 merged, I believe this can be addressed.

I'm glad that were handling this now! I noticed a major slowdown here too when adjusting the form URL, and was thinking it was related to the `onChange` event. Looking over the code limiting the amount of re-rendering and testing the branch, it appears everything works well.

**Comment by tuckerzp** [comment]
It looks like the `oscalUrl` is not actually being changed to the new `value` here? I am unsure why the `setter` of something initialized from `useState` would lag behind one value.

https://github.com/EasyDynamics/oscal-react-library/blob/feature/make-form-behave-predictably/src/components/OSCALLoader.js#L130-#L132

**Comment by tuckerzp** [comment]
I will wait for @kylelaker to review my change and take it out of draft, but I think this should be fixed. There was a weird issue where we were reloading before the state had a change to actually update.

File: src/components/OSCALLoaderForm.js
Comment (kylelaker) [inline src/components/OSCALLoaderForm.js:16]: I am wondering if we actually need this `useRef`. I added it as a wild guess. But I am curious if we can remove it?

Comment (tuckerzp) [reply src/components/OSCALLoaderForm.js:16]: After playing around with removing it a bit, it seems that `useRef` may be needed. At the very least I am having trouble getting it to work without it.

**Comment by kylelaker** [comment]
Thanks for helping with this @tuckerzp. I approve of the changes you've made here; I think you should feel free to mark this PR as approved as well (if you do approve).

**Review by tuckerzp — APPROVED** [review]

**Review by mikeisen1 — APPROVED** [review]

**Pull request merged**
